packages/plugins/Web3Profile/src/SNSAdaptor/components/FollowLensDialog.tsx

```diff
@@ -0,0 +1,287 @@
+
+        if (!profile) return
+        const isFollowing = await Lens.queryFollowStatus(account, profile.id)
+        const defaultProfile = await Lens.queryDefaultProfileByAddress(account)
+
+        const profiles = await Lens.queryProfilesByAddress(account)
+
+        return {
+            profile,
+            isFollowing,
+            defaultProfile: defaultProfile ?? first(profiles),
+        }
+    }, [handle, open, account])
+
+    const { isFollowing, profile, defaultProfile } = value ?? {}
+
+    const followModule = useMemo(() => {
+        if (profile?.followModule?.type === FollowModuleType.ProfileFollowModule && defaultProfile) {
+            return {
+                profileFollowModule: {
+                    profileId: defaultProfile.id,
+                },
+            }
+        } else if (profile?.followModule?.type === FollowModuleType.FeeFollowModule && profile.followModule.amount) {
+            return {
+                feeFollowModule: {
+                    currency: profile.followModule.amount.asset.address,
+                    value: profile.followModule.amount.value,
+                },
+            }
+        }
+        return
+    }, [profile, defaultProfile])
+    // #endregion
+
+    // #region follow and unfollow event handler
+    const [{ loading: followLoading }, handleFollow] = useFollow(profile?.id, followModule, retry)
+    const [{ loading: unfollowLoading }, handleUnfollow] = useUnfollow(profile?.id, retry)
+    // #endregion
+
+    const { value: feeTokenBalance, loading: getBalanceLoading } = useFungibleTokenBalance(
+        NetworkPluginID.PLUGIN_EVM,
+        profile?.followModule?.amount?.asset.address ?? '',
+    )
+
+    const disabled = useMemo(() => {
+        if (
+            !!wallet?.owner ||
+            chainId !== ChainId.Matic ||
+            followLoading ||
+            unfollowLoading ||
+            (profile?.followModule?.type === FollowModuleType.ProfileFollowModule && !defaultProfile) ||
+            (profile?.followModule?.type === FollowModuleType.FeeFollowModule &&
+                profile.followModule.amount &&
+                (!feeTokenBalance ||
+                    isLessThan(
+                        formatBalance(feeTokenBalance, profile.followModule.amount.asset.decimals),
+                        profile.followModule.amount.value,
+                    ))) ||
+            profile?.followModule?.type === FollowModuleType.RevertFollowModule
+        )
+            return true
+        return false
+    }, [wallet?.owner, chainId, followLoading, unfollowLoading, feeTokenBalance, profile?.followModule])
+
+    const [element] = useHover((isHovering) => {
+        const getButtonText = () => {
+            if (isFollowing) {
+                return isHovering ? t.unfollow() : t.following_action()
+            } else if (
+                profile?.followModule?.type === FollowModuleType.FeeFollowModule &&
+                profile.followModule.amount
```

Comment:
isZero

Reply:
amount is not a number value, it is an object.